spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -842,11 +853,24 @@ private static boolean hasReturnParameterizedWithMessage(Method method, boolean
 		return false;
 	}
 
+	private boolean hasReturnMessageTypeOnFunction(Method method) {
+		if (Function.class.isAssignableFrom(this.serviceInterface) && "apply".equals(method.getName())) {
```

Comment:
Wondering if NOT `&& "apply".equals(method.getName())` should really trigger the exception instead of simply returning `false`? I mean that would be some gross miss-configuration, right?

Reply:
Your interface may `extends Function`, but have more methods declared.
Why is that so invalid?

Reply:
What I am saying is that at that moment it is really not a Function. What I mean is that you may extend and have some additional methods, but in the Functional world the only method I know is `apply` and that is my entry point if I down-cast everything to Function.
It may not be a problem in your context, but i wanted to bring it up to ensure you thought of that.

Reply:
OK. Thanks for your feedback, but for extra safety I would stay with this extra protection.